File: 2021/12/26/main.py
class Solution:
    def kClosest(self, points: List[List[int]], k: int) -> List[List[int]]:
        def dist(x):
            return x[0] * x[0] + x[1] * x[1]

        return heapq.nsmallest(k, points, key=lambda x: dist(x))

File: 2021/12/26/main.go
package main

import "sort"

func kClosest(points [][]int, k int) [][]int {
	sort.Slice(points, func(i int, j int) bool {
		return dist(points[i]) < dist(points[j])
	})
	return points[:k]
}

func dist(p []int) int {
	return p[0]*p[0] + p[1]*p[1]
}
